addons/website_livechat/static/src/js/im_livechat.js

```diff
@@ -10,18 +10,15 @@ LivechatButton.include({
 
     /**
      * @override
-     * This will will correctly format the livechat session cookie
-     * that comes from server side (and that is not properly formatted)
+     * Check if a chat request is opened for this visitor
+     * if yes, replace the session cookie and start the conversation immediately.
+     * Do this before calling super to have everything ready before executing existing start logic.
      * This is used for chat request mechanism, when an operator send a chat request
      * from backend to a website visitor.
      */
     willStart: function () {
-        var self = this;
-        var cookie = utils.get_cookie('im_livechat_session');
-        var ready;
-        if (cookie) {
-            var cleanedLivechatSessionCookie = this.decode_server_cookie(cookie);
-            utils.set_cookie('im_livechat_session', cleanedLivechatSessionCookie, 60*60);
+        if (this.options.chat_request_session) {
+            utils.set_cookie('im_livechat_session', JSON.stringify(this.options.chat_request_session), 60*60);
```

Comment:
You do realize that you move from: `expiration_date = datetime.now() + timedelta(days=100 * 365)  # never expire`
to: `60*60`

Seems strange.